cmd/dump.go

```diff
@@ -78,18 +78,37 @@ func dumpInstalledObjectsCmdAction(cmd *cobra.Command, args []string) error {
 		return cobraext.FlagParsingError(err, cobraext.DumpOutputFlagName)
 	}
 
-	tlsSkipVerify, _ := cmd.Flags().GetBool(cobraext.TLSSkipVerifyFlagName)
-
-	var clientOptions []elasticsearch.ClientOption
-	if tlsSkipVerify {
-		clientOptions = append(clientOptions, elasticsearch.OptionWithSkipTLSVerify())
+	tlsSkipVerify, err := cmd.Flags().GetBool(cobraext.TLSSkipVerifyFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.TLSSkipVerifyFlagName)
 	}
 
 	profile, err := cobraext.GetProfileFlag(cmd)
 	if err != nil {
 		return err
 	}
 
+	var kibanaOptions []kibana.ClientOption
+	if tlsSkipVerify {
+		kibanaOptions = append(kibanaOptions, kibana.TLSSkipVerify())
+	}
+	kibanaClient, err := stack.NewKibanaClientFromProfile(profile, kibanaOptions...)
+	if err != nil {
+		return fmt.Errorf("failed to initialize Kibana client: %w", err)
+	}
+	installedPackage, err := kibanaClient.GetPackage(packageName)
+	if err != nil {
+		return fmt.Errorf("failed to get package status: %w", err)
+	}
+	if installedPackage.Status == "not_installed" {
+		cmd.Printf("Package %s is not installed.\n", packageName)
+		return nil
+	}
```

Comment:
When trying this in a Elastic stack 7.17* it does not work as expected.

In this test, `mysql_enteprise` is not installed, but it does not show the message `Package mysql_enterprise is not installed.`

```
 $ elastic-package stack up -v -d --version 7.17.18
 $ elastic-package dump installed-objects -P mysql_enterprise
Error: no objects found for package mysql_enterprise

```

The JSON in the response is a bit different in 7.17, instead of `item` it has a `response` field.

```
 $ elastic-package stack up -v -d --version 7.17.18
 $ curl -s -k -u elastic:changeme https://localhost:5601/api/fleet/epm/packages/mysql_enterprise | jq -r .response.status
not_installed

 $ curl -s -k -u elastic:changeme https://localhost:5601/api/fleet/epm/packages/mysql_enterprise | jq -r .item.status
null
```

In a Elastic stack 8.12.1 , the JSON response contains both fields `response` and ìtem`:

```
 $ elastic-package stack up -v -d --version 8.12.1
 $ curl -s -k -u elastic:changeme https://localhost:5601/api/fleet/epm/packages/mysql_enterprise | jq -r .item.status
not_installed

 $ curl -s -k -u elastic:changeme https://localhost:5601/api/fleet/epm/packages/mysql_enterprise | jq -r .response.status
not_installed
```

Reply:
Should it be added that logic for 7.17.* ?

Maybe, it's not needed. If the `mysql_enterprise` is installed in 7.17.18, the command does dump the expected installed objects:

```
 $ elastic-package stack up -v -d --version 7.17.18
 $ elastic-package dump installed-objects -P mysql_enterprise
Error: no objects found for package mysql_enterprise

 # install package
 $ elastic-package install -v 

 $ elastic-package dump installed-objects -P mysql_enterprise
Dumped 7 installed objects for package mysql_enterprise to package-dump
```

so, at the end it does dump the objects/files if package exists/is installed.

Reply:
Thanks, I didn't remember this difference, added support for 7.17 in https://github.com/elastic/elastic-package/pull/1674/commits/1e821620fb7c4edd18ed30d94eeda6b361d665ca